flake.nix

```diff
@@ -0,0 +1,61 @@
+{
+  description = "Constellation";
+
+  inputs = {
+    nixpkgsWorking = {
+      url = "github:katexochen/nixpkgs/working";
+    };
+    nixpkgsUnstable = {
+      url = "github:NixOS/nixpkgs/nixpkgs-unstable";
+    };
+    flake-utils = {
+      url = "github:numtide/flake-utils";
+    };
+  };
+
+  outputs =
+    { self
+    , nixpkgsWorking
+    , nixpkgsUnstable
+    , flake-utils
+    }:
+    flake-utils.lib.eachDefaultSystem (system:
+    let
+      pkgsWorking = import nixpkgsWorking { inherit system; };
+      pkgsUnstable = import nixpkgsUnstable { inherit system; };
+
+      mkosiDev = (pkgsWorking.mkosi.overrideAttrs (oldAttrs: rec {
+        src = pkgsWorking.fetchFromGitHub {
+          owner = "systemd";
+          repo = "mkosi";
+          rev = "a9ebc1e556e64728f1b508b43a15b065f2cdc7ca";
+          hash = "sha256-879xxfjYoH6OV5OpJVeeePi+cbT86c4vzrJ3vLvik+0=";
+        };
+        propagatedBuildInputs = oldAttrs.propagatedBuildInputs ++ (with pkgsWorking;  [
+          # package management
+          dnf5
+          rpm
+
+          # filesystem tools
+          squashfsTools # mksquashfs
+          dosfstools # mkfs.vfat
+          mtools # mcopy
+          cryptsetup # dm-verity
+          util-linux # flock
+          kmod # depmod
+        ]);
+        # TODO(malt3): re-enable config tests once the distribution detection is fixed
+        disabledTestPaths = [
+          "tests/test_config.py"
+        ];
+      }));
+
+    in
+    {
+      packages.mkosi = mkosiDev;
+
+      devShells.default = import ./nix/shells/default.nix { pkgs = pkgsUnstable; };
+
+      formatter = nixpkgsUnstable.legacyPackages.${system}.nixpkgs-fmt;
```

Comment:
We should add a formatter/linter/instantiate workflow to the CI (not in this PR, but soon).